python/ray/data/dataset.py

```diff
@@ -4203,6 +4203,12 @@ def __len__(self) -> int:
             "This may be an expensive operation."
         )
 
+    def __iter__(self):
+        raise TypeError(
+            "`Dataset` objects aren't iterable. If you want to inspect records, call "
```

Comment:
would it be better to mention the iteration APIs instead of `take`?

```
`Dataset` objects aren't iterable. If you want to iterate records, call `ds.iter_rows()` or `ds.iter_batches()`.
See more information at https://docs.ray.io/en/latest/data/consuming-datasets.html
```

Reply:
Yeah, I think that'd make more sense. Will update the message.